src/playlist-controller.js

```diff
@@ -933,17 +933,9 @@ export class PlaylistController extends videojs.EventTarget {
 
     this.switchMedia_(media, 'fast-quality');
 
-    // Delete all buffered data to allow an immediate quality switch, then seek to give
-    // the browser a kick to remove any cached frames from the previous rendtion (.04 seconds
-    // ahead was roughly the minimum that will accomplish this across a variety of content
-    // in IE and Edge, but seeking in place is sufficient on all other browsers)
-    // Edge/IE bug: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/14600375/
-    // Chrome bug: https://bugs.chromium.org/p/chromium/issues/detail?id=651904
-    this.mainSegmentLoader_.resetEverything(() => {
-      this.tech_.setCurrentTime(this.tech_.currentTime());
-    });
-
-    // don't need to reset audio as it is reset when media changes
+    // Reset main segment loader properties and next segment position information
+    this.mainSegmentLoader_.resetLoaderProperties();
+    this.mainSegmentLoader_.resetLoader();
```

Comment:
I see that `resetLoaderProperties` resets `initSegment` things + `ended`, should we make it a part of the `resetLoader`?

Reply:
That was my original inclination, however `resetLoader` is called in a couple other spots so I decided against it for now. Definitely open to moving it in there if it makes sense, the risk seems rather low given the other uses.

Reply:
Make sense, 
maybe we can leave a comment about this?

Reply:
Good idea. Also, looking at this again I think `// don't need to reset audio as it is reset when media changes` is still useful. Going to add that back in.